src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -724,7 +724,20 @@ void BaseCouplingScheme::advanceTXTWriters()
 
 bool BaseCouplingScheme::reachedEndOfTimeWindow()
```

Comment:
```suggestion
bool BaseCouplingScheme::reachedEndOfTimeWindow() const
```
Actually, this should be a const function. I would suggest to do all the non-const stuff that modifies `_computedTimeWindowPart` or `data` via `data->setSampleAtTime` at a different place.

Reply:
Sorry for making things more complicated, but I think the `const` was really missing in the first place. I directly added it on `develop`. See 6ee29277f892e6d08a514fa734f1274aa8b59cea. Please don't forget to merge again.